src/PrestaShopBundle/Command/UpdateLicensesCommand.php

```diff
@@ -178,21 +264,44 @@ private function addLicenceToNode($node, $file)
         $comments = $node->getAttribute('comments');
         foreach ($comments as $comment) {
             if ($comment instanceof \PhpParser\Comment
-                    && strpos($comment->getText(), 'prestashop') !== false) {
-                file_put_contents($file->getRelativePathname(), str_replace($comment->getText(), $this->text, $file->getContents()));
+                && strpos($comment->getText(), 'prestashop') !== false) {
+                file_put_contents($file->getRelativePathname(), str_replace($comment->getText(), $this->license, $file->getContents()));
             }
         }
     }
 
-    private function addLicenceToSmartyTemplate($file)
+    /**
+     * @param SplFileInfo $file
+     */
+    private function addLicenseToSmartyTemplate(SplFileInfo $file)
     {
-        $this->addLicenceToFile($file, '{', '}');
+        $this->addLicenseToFile($file, '{', '}');
     }
 
-    private function addLicenceToTwigTemplate($file)
+    /**
+     * @param SplFileInfo $file
+     */
+    private function addLicenseToTwigTemplate(SplFileInfo $file)
     {
         if (strrpos($file->getRelativePathName(), 'html.twig') !== false) {
-            $this->addLicenceToFile($file, '{#', '#}');
+            $this->addLicenseToFile($file, '{#', '#}');
         }
     }
+
+    /**
+     * @param SplFileInfo $file
+     * @return bool
+     */
+    private function addLicenseToJsonFile(SplFileInfo $file)
+    {
+        if (!in_array($file->getFilename(), array('composer.json', 'package.json'))) {
+            return false;
+        }
+
+        $content = (array) json_decode($file->getContents());
```

Comment:
This works for the first dimension, but may fail in case of multidimensional arrays.

If you meet some issues with `json_decode`, think about the second parameters `$assoc` (http://php.net/manual/en/function.json-decode.php). :wink: